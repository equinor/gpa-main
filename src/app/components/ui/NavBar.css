@charset "UTF-8";
@font-face {
  font-family: Equinor;
  src: local("â˜»"), url(/fonts/Equinor-Regular.woff2) format("woff2"), url(/fonts/assets/fonts/Equinor-Regular.woff) format("woff");
  font-style: normal;
  font-weight: 400;
}

.NavBar {
  width: 100%;
  background: #FFFFFF;
  float: left;
  height: 64px;
  border-bottom: 2px solid #f7f7f7;
}

.NavBar .left {
  float: left;
  margin: 0 0 0 50px;
}

.NavBar .left span {
  font-family: "Equinor";
  color: black;
  display: block;
  float: left;
  margin: 25px 0 0 25px;
  font-size: 16px;
}

.NavBar svg {
  color: #007079;
  width: 24px;
  height: 24px;
  margin: 20px 0 0 0;
  float: left;
}

.NavBar .right {
  float: right;
  margin: 0 50px 0 0;
}
/*# sourceMappingURL=NavBar.css.map */