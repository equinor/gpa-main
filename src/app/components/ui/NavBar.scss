@import "../../utils/css/branding/brandingDefinitions.scss";

.NavBar {
    width: 100%;
    background: #FFFFFF;
    float: left;
    height: 64px;
    border-bottom: 2px solid #f7f7f7;

    .left {
        float: left;
        margin: 0 0 0 50px;

        span {
            font-family: "Equinor";
            color: black;
            display: block;
            float: left;
            margin: 25px 0 0 25px;
            font-size: 16px;
        }
    }

    svg {
        color: $green;
        width: 24px;
        height: 24px;
        margin: 20px 0 0 0;
        float: left;
    }

    .right {
        float: right;
        margin: 0 50px 0 0;
    }

}