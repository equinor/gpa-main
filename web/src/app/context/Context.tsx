import React, { createContext, useContext, useReducer } from 'react';

var a: any;
export const StateContext = createContext(a);

export const StateProvider = ({ children, accountInfo }: any) => {
    return (
    <StateContext.Provider 
        //initial reducer values
        value={useReducer(appReducer, {
            authentificated: true,
            accountInfo: {
                userName: accountInfo.userName,
                name: accountInfo.name
            }
        })}    
    >
        {children}
    </StateContext.Provider>
    )
};

export const useStateContext = () => useContext(StateContext);

//reducer
export interface IAppState {
    authentificated: boolean;
    accountInfo: {
        userName: string,
        name: string
    };
}

export const appState = {
    authentificated: false,
    accountInfo: {}
};

export enum EReducerActions {
}

export const appReducer = (state: IAppState, action: {
    type: EReducerActions,
    payload: any
}) => {
    switch (action.type) {
        default:
            return state;
    }
};