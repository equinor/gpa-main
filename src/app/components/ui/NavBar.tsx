import './NavBar.css';

import React from "react";
import { EquinorIcon } from '../../assets/svg/EquinorIcon';

export interface IProps { }

export interface IState { }

export class NavBar extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
    };
  }

  render() {
    return (
      <div className="NavBar">
        <div className="left">
          <EquinorIcon icon="menu"></EquinorIcon>
          <span>Neqsim</span>
        </div>
        <div className="right">
          <EquinorIcon icon="user"></EquinorIcon>
        </div>
      </div> 
    );
  }
};
