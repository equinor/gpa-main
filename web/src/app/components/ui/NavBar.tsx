import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components/macro';

import { EIcon, EquinorIcon } from '../../assets/svg/EquinorIcon';
import { IAppState, useStateContext } from '../../context/Context';

export const NavBar: React.FunctionComponent = () => {
  const [{accountInfo}] : [IAppState] = useStateContext();
  
  return (
    <StNavBar>
      <StHeader>
        {/* Logo */}
        {/* <Link to="/neqsim">
          <img className="logo" src="./logo.png"></img>
        </Link> */}
        <Link to="/neqsim">
          <EquinorIcon icon={EIcon.MENU} size={24} />
        </Link>
        <span>
          <Link to="/neqsim">Neqsim</Link>
        </span>
      </StHeader>
      <div style={{ marginRight: '50px', height: "24px" }}>
        {accountInfo.name}
      </div>
    </StNavBar >
  );
};

const StNavBar = styled.nav`
      display: flex;
      flex-direction: row;
      width: 100%;
      min-height: 64px;
      border-bottom: 2px solid #f7f7f7;
      align-items: center;
      justify-content: space-between;
    `;

const StHeader = styled.div`
      display: flex;
      flex-direction: row;
      margin-left: 50px;
      align-items: center;
      font-family: Equinor, serif;
  img.logo {
        width: 80px;
      height: 55px;
      margin: 3px 0;
      object-fit: cover;
    }
  > span {
        margin: 2px 0 0 25px;
      font-size: 18px;
    font-size: 18px; 
      font-size: 18px;
      font-weight: 500;
      cursor: pointer;
    > a, a:hover {
        color: black;
      text-decoration: none
    }
  }
`;