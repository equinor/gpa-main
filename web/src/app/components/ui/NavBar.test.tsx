import { MockedProvider } from '@apollo/react-testing';
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { NavBar } from './NavBar';
import { StateProvider } from '../../context/Context';


describe("Navbar", () => {
    it("snapshots", () => {
        const tree = renderer.create(
            <StateProvider accountInfo={
             {
                 userName: "testUserName",
                 name: "testName"
             }   
            }>
                <MockedProvider addTypename={false}>
                    <Router>
                        <NavBar />
                    </Router>
                </MockedProvider>
            </StateProvider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
});  