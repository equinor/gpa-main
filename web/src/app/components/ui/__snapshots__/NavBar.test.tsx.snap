// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshots 1`] = `
<nav
  className="NavBar__StNavBar-qlt09m-0 fbmlKN"
>
  <div
    className="NavBar__StHeader-qlt09m-1 cEJAzB"
  >
    <a
      href="/neqsim"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M3 8V6H21V8H3ZM3 13H21V11H3V13ZM3 18H21V16H3V18Z"
          fill="#007079"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <span>
      <a
        href="/neqsim"
        onClick={[Function]}
      >
        Neqsim
      </a>
    </span>
  </div>
  <div
    style={
      Object {
        "height": "24px",
        "marginRight": "50px",
      }
    }
  >
    testName
  </div>
</nav>
`;
